src/main/scala/chisel3/util/Reg.scala

```diff
@@ -124,6 +124,96 @@ object ShiftRegister {
     compileOptions:      CompileOptions
   ): T =
     ShiftRegisters(in, n, resetData, en).lastOption.getOrElse(in)
+
+  /** Returns the n-cycle delayed version of the input signal (SyncReadMem-based ShiftRegister implementation).
+    *
+    * @param in input to delay
+    * @param n  number of cycles to delay
+    * @param en enable the shift
+    * @param useDualSRAMPort dual port or single port SRAM based implementation
+    * @param name name of SyncReadMem object
+    */
+  def mem[T <: Data](in: T, n: Int, en: Bool, useDualSRAMPort: Boolean, name: Option[String]): T =
+    macro SourceInfoTransform.inNEnUseDualSRAMpNameArg
+
+  /** @group SourceInfoTransformMacro */
+  def do_mem[T <: Data](
+    in:              T,
+    n:               Int,
+    en:              Bool,
+    useDualSRAMPort: Boolean,
+    name:            Option[String]
+  )(
+    implicit sourceInfo: SourceInfo,
+    compileOptions:      CompileOptions
+  ): T = _apply_impl_mem(in, n, en, useDualSRAMPort, name)
+
+  private def _apply_impl_mem[T <: Data](
+    in:              T,
+    n:               Int,
+    en:              Bool = true.B,
+    useDualSRAMPort: Boolean = false,
```

Comment:
and so on...

Reply:
Changed all occurrences of `useDualSRAMPort` to `useDualPortSram`, as you've suggested. I also think it looks a bit better with the new parameter name.